app/models/user.rb

```diff
@@ -8,6 +8,7 @@ class User < ActiveRecord::Base
   has_many :posts
   before_save :check_if_first_user
   has_many :comments 
+  validates :name, presence: { :message => "Name is required!" }
 
```

Comment:
It would be useful to build a named scope to get all users who should be notified.  I think something like this:

```
scope :users_to_notify, -> { where(:post_notification => true)  }
```

We could hold off on doing so until we combine both your code and the code @ceceliahavens is working on though.

http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope